service/header/interface.go

```diff
@@ -0,0 +1,26 @@
+package header
+
+import "context"
+
+// Subscriber encompasses the behavior necessary for a node to
+// subscribe/unsubscribe from new ExtendedHeader events from the
+// network.
+type Subscriber interface {
+	Subscribe(ctx context.Context) (<-chan *ExtendedHeader, error)
+	Unsubscribe() error
+}
+
+// Exchange encompasses the behavior necessary for a node to
+// request ExtendedHeaders and respond to ExtendedHeader requests
+// from the network.
+type Exchange interface {
+	RequestHeaders(ctx context.Context, request *ExtendedHeaderRequest) ([]*ExtendedHeader, error)
+	RespondToHeadersRequest(ctx context.Context, request *ExtendedHeaderRequest) error
+}
```

Comment:
This design misses the code that receives ExtendedHeaderRequests to respond using RespondToheadersRequest or which part here should listen for incoming requests?

---
The other more simplified design for this interface can look like this:
```go
	Request(ctx context.Context, heaiht int64) (*ExtendedHeaders, error)
	RequestRange(ctx context.Context, from, to int64) ([]*ExtendedHeaders, error)
```

Here we don't expose any protocol-level details and just expose a simple API to request things. Internally, we will serve requests for other peers. Specifically, there will be a goroutine in the protocol that listens for incoming requests and responds with error or success depending on the Store's possesity of the requested Headers. Again, this should happen inside in the Exchange and not outside.